Change API error for invalid langs to JSON

kolakachi opened the pull request:
Added a try and catch block in apiAction in OcrController, Catch block returns a json response.

Bug: T330243

kolakachi committed:
Change API error for invalid langs to JSON

Added a try and catch block in apiAction in OcrController,
Catch block returns a json response.

Bug: T330243

kolakachi committed:
Fix linting errors

Parthiv-M reviewed (commented):
Worked fine for me

samwilson commented on src/Controller/OcrController.php:
```diff
@@ -220,16 +221,22 @@ public function homeAction(): Response
+            return $this->getApiResponse($responseParams);
+        } catch (Exception $exception) {
+            return $this->getApiResponse([
+                "message" => $exception->getMessage(),
```
Other error states return the error in the `error` key, so this should probably do the same, rather than `message`.

samwilson commented on src/Controller/OcrController.php:
```diff
@@ -220,16 +221,22 @@ public function homeAction(): Response
-        if ($warnings) {
-            $responseParams['warnings'] = $warnings;
+        try {
+            $this->setup();
```
The exception is actually only coming from this method call isn't it? It might be better to just try/catch around this then, because the other calls below already have functioning error handling.

samwilson reviewed (changes requested):
It looks like this breaks the error output of requests such as `/api?langs[]=foobar`, which now returns only `{"message":""}` instead of the error message and other info.

kolakachi committed:
Update apiAction method

Change response object key from message to error
Move try/catch block to cover setup method only.

Bug: T330243
